tests/test_patch.py

```diff
@@ -751,3 +751,23 @@ def sample():
         """)
 
     assert sample() == 15 * 4
+
+
+def test_patch_by_path(tmpdir):
+    tmpdir.join('patch_by_path.py').write(dedent("""\
+        class Foo(object):
+            def sample(self):
+                return 1
+        """))
+    sys.path.insert(0, six.text_type(tmpdir))
+    try:
+        patchy.patch('patch_by_path.Foo.sample', """\
+            @@ -2,2 +2,2 @@
+            -    return 1
+            +    return 2
+            """)
+        from patch_by_path import Foo
```

Comment:
Worth testing both cases where the caller has and has not already imported Foo?

Reply:
done